### DISCUSSION TITLE
data enhancements like logstash

### DISCUSSION THREAD

**Discussion by hzhuchao** [vectordotdev/vector#16662] [Q&A] [answered]
such as the jdbc connection to an external database in vector tansforms . for example logstash‘s elasticsearch filter or jdbc_static filter.  reference url  https://www.elastic.co/guide/en/logstash/current/lookup-enrichment.html

**Comment by StephenWakely** [comment, marked as answer]
The primary issue that we are facing when creating enrichments from external sources is performance. With something like an external database each call would involve a network round trip. Currently all calls within remap are synchronous, introducing networking into the remap process would mean we would have to change the structure of remap to include asynchronous calls.

It is definitely something we have considered, but we want to make sure that it is something we get right. Currently there hasn't been enough user demand for us to consider actually adding it to our immediate roadmap. 

Please feel free to add a feature request here https://github.com/vectordotdev/vector/issues and include as much context as you can. This will give us more information about how we can take this further.
